src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerAssemblyLoader.Core.cs

```diff
@@ -83,6 +96,60 @@ private partial Assembly Load(AssemblyName assemblyName, string assemblyOriginal
             return loadContext.LoadFromAssemblyName(assemblyName);
         }
 
+        /// <summary>
+        /// Is this a registered analyzer file path that the loader knows about.
+        /// 
+        /// Note: this is using resolved paths, not the original file paths
+        /// </summary>
+        private bool IsRegisteredAnalyzerPath(string analyzerPath)
+        {
+            CheckIfDisposed();
+
+            lock (_guard)
+            {
+                return _resolvedToOriginalPathMap.ContainsKey(analyzerPath);
+            }
+        }
+
+        internal string? GetAssemblyLoadPath(AssemblyName assemblyName, string directory)
+        {
+            // Prefer registered dependencies in the same directory first.
+            var simpleName = assemblyName.Name!;
+            var assemblyPath = Path.Combine(directory, simpleName + ".dll");
+            if (IsRegisteredAnalyzerPath(assemblyPath))
+            {
+                return assemblyPath;
+            }
+
+            // Next if this is a resource assembly for a known assembly then load it from the 
+            // appropriate sub directory if it exists
+            //
+            // Note: when loading from disk the .NET runtime has a fallback step that will handle
+            // satellite assembly loading if the call to Load(satelliteAssemblyName) fails. This
+            // loader has a mode where it loads from Stream though and the runtime will not handle
+            // that automatically. Rather than bifurcate our loading behavior between Disk and
+            // Stream both modes just handle satellite loading directly
+            if (assemblyName.CultureInfo is not null && simpleName.EndsWith(".resources", SimpleNameComparer.Comparison))
+            {
+                var analyzerFileName = Path.ChangeExtension(simpleName, ".dll");
+                var analyzerFilePath = Path.Combine(directory, analyzerFileName);
+                return GetSatelliteLoadPath(analyzerFilePath, assemblyName.CultureInfo);
```

Comment:
Why don't we consult GetBestResolvedPath for satellite assemblies here but we do in the Desktop counterpart?

Reply:
In the core case we know the directory where the assembly is loaded from so we don't have to use `GetBestResolvedPath` to _guess_ at it.